### PR TITLE
migrate c-blosc to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This PR was auto-generated by a bash script. Most of the recipes (around 80%) can be built. It this one is failing, feel free to contribute a manual fix.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/c-blosc/all/conanfile.py [hunk 1]
```diff
@@ -84,9 +85,9 @@ def build(self):
 
     def _patch_sources(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+            tools.files.patch(self, **patch)
```
Comment (uilianries) [inline recipes/c-blosc/all/conanfile.py:88]: prefer `apply_conandata_patches` instead.

https://docs.conan.io/en/latest/reference/conanfile/tools/files/patches.html#conan-tools-files-apply-conandata-patches

File: recipes/c-blosc/all/test_package/conanfile.py [hunk 2]
```diff
@@ -12,6 +13,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
+        if not tools.build.cross_building(self, self):
```
Comment (ericLemanissier) [inline recipes/c-blosc/all/test_package/conanfile.py:16]:
```suggestion
        if not tools.build.cross_building(self):
```

Comment (AndreyMlashkin) [reply recipes/c-blosc/all/test_package/conanfile.py]: I have fixed this in later script version

**Comment by uilianries** [comment]
@AndreyMlashkin Please, stopping using your script on CCI for testing, it consumes resources and CI time, which could be used for others PR waiting in the queue.

Thank you that you are creating a tool to make our life easier, but you have to mature it **locally** first. Test it in your local environment, fix/improve, then you can use here.

All errors so far, could be spotted with a single `conan export` in your machine.

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin Please, stopping using your script on CCI for testing, it consumes resources and CI time, which could be used for others PR waiting in the queue.

Yes, I have a local check starting from script version 4.
conan create does not return a code, so I need to relay on grep output, if a test_recipe was compiled.

Anyway, there are less than 25 PR created

**Comment by ericLemanissier** [comment]
> conan create does not return a code

it does return a code

**Comment by AndreyMlashkin** [comment]
> it does return a code. You probably miss `set -e` and `set -o pipefail`

```
        conan create . $full_recipe_name --build missing -ks > local_build.txt
        echo "return code is $?"
```
For a regular failed build and a succeeded recipe, it will be printed "return code is 0"
It returns a different value for conan exception and some other conditions

**Comment by ericLemanissier** [comment]
```
eric@DESKTOP-1HV1KQ7:~$ conan create /mnt/c/pf/conan-center-index/recipes/gettext/all/ gettext/0.20.1@ > tmp.txt
ERROR: Error loading conanfile at '/mnt/c/pf/conan-center-index/recipes/gettext/all/conanfile.py': Unable to load conanfile in /mnt/c/pf/conan-center-index/recipes/gettext/all/conanfile.py
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/mnt/c/pf/conan-center-index/recipes/gettext/all/conanfile.py", line 5, in <module>
    import blaoiauzerpoiu
ModuleNotFoundError: No module named 'blaoiauzerpoiu'

eric@DESKTOP-1HV1KQ7:~$ echo $?
1
```

**Review by ericLemanissier — DISMISSED** [review]

**Review by danimtb — APPROVED** [review]
Is a step forward towards conan 2 support but the recipe still requires some work. I am fine with that, but for next PRs I'd prefer them to be more complete to avoid wasting CI resources and generating extra recipe revisions

File: recipes/c-blosc/all/conanfile.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile, tools
+from conan.tools.files import apply_conandata_patches
+from conans import CMake
 import os
 
 required_conan_version = ">=1.36.0"
```
Comment (SpaceIm) [inline recipes/c-blosc/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.47.0"
```

**Comment by SpaceIm** [comment]
Let's merge this one, but I'll submit a full migration anyway, so I don't know whether it's worth to create a RREV for such change.

**Review by SpaceIm — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
The more complete the better, but every step is valuable. This PR introduces the changes required to be green, so it is a step forward. New PRs to this recipe (even newbies) will start with something that works, they don't need to fix unrelated changes required by linters.

**Review by danimtb — APPROVED** [review]

**Pull request merged**